pkg/api/handlers/compat/events.go

```diff
@@ -15,77 +16,127 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// filtersFromRequests extracts the "filters" parameter from the specified
+// http.Request.  The paramater can either be a `map[string][]string` as done
+// in new versions of Docker and libpod, or a `map[string]map[string]bool` as
+// done in older versions of Docker.  We have to do a bit of Yoga to support
+// both - just as Docker does as well.
+//
+// Please refer to https://github.com/containers/podman/issues/6899 for some
+// background.
+func filtersFromRequest(r *http.Request) []string {
+	var (
+		compatFilters map[string]map[string]bool
+		filters       map[string][]string
+		libpodFilters []string
+	)
+	raw := []byte(r.Form.Get("filters"))
+
+	// Backwards compat with older versions of Docker.
+	if err := json.Unmarshal(raw, &compatFilters); err == nil {
+		for filterKey, filterMap := range compatFilters {
+			for filterValue, toAdd := range filterMap {
+				if toAdd {
+					libpodFilters = append(libpodFilters, fmt.Sprintf("%s=%s", filterKey, filterValue))
+				}
+			}
+		}
+		return libpodFilters
+	}
+
+	if err := json.Unmarshal(raw, &filters); err == nil {
+		for filterKey, filterSlice := range filters {
+			for _, filterValue := range filterSlice {
+				libpodFilters = append(libpodFilters, fmt.Sprintf("%s=%s", filterKey, filterValue))
+			}
+		}
+		return libpodFilters
+	}
+
+	return nil
```

Comment:
This function should return an err correct?  If the second unmarshal fails, we got bad data

Reply:
Good catch, yes! I'll wrap it up and repush in a bit.